src/tests/unit/auto/get_device_list.cpp

```diff
@@ -0,0 +1,124 @@
+// Copyright (C) 2018-2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
```

Comment:
should we also add per plugin tests that `AUTO:-<DEVICE_NAME>` does not have device?

BTW, what will happen for AUTO:-CPU if only CPU is available on machine?

Reply:
> Hi @songbell @wangleis can we reflect this change in the documentation as well? with the description why such syntax can be needed? (personally, I hear about such requirement for the first time :) ) Thanks!

Sure. Will update AUTO document for 2022.2. The requirement is coming from CCG and PAE.

Reply:
> should we also add per plugin tests that `AUTO:-<DEVICE_NAME>` does not have device?
> 
> BTW, what will happen for AUTO:-CPU if only CPU is available on machine?

If CPU is only available device on machine, AUTO:-CPU will throw exception due to no available candidate device